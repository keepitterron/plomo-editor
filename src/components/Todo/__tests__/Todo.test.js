import React from 'react';
import { Todo } from '../index.js';
import { shallow, mount } from 'enzyme';

describe('Todo', () => {
  it('renders a basic todo', () => {
    const blockProps = {onToggle: 'foo', checked: false};
    const component = shallow(<Todo offsetKey="foo" blockProps={blockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('renders a checked todo', () => {
    const blockProps = {onToggle: 'foo', checked: true};
    const component = shallow(<Todo offsetKey="foo" blockProps={blockProps} />);
    expect(component).toMatchSnapshot();
  });
});
