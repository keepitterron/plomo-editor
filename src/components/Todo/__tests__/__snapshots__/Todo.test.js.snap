// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders a basic todo 1`] = `
<div
  className="todo-list-item__container"
  data-offset-key="foo"
>
  <div
    className="todo-list-item__checkbox"
    contentEditable={false}
    suppressContentEditableWarning={true}
  >
    <input
      checked={false}
      onChange="foo"
      type="checkbox"
    />
  </div>
  <div
    className="todo-list-item__text"
  >
    <DraftEditorBlock
      blockProps={
        Object {
          "checked": false,
          "onToggle": "foo",
        }
      }
      offsetKey="foo"
    />
  </div>
</div>
`;

exports[`Todo renders a checked todo 1`] = `
<div
  className="todo-list-item__container is-checked"
  data-offset-key="foo"
>
  <div
    className="todo-list-item__checkbox"
    contentEditable={false}
    suppressContentEditableWarning={true}
  >
    <input
      checked={true}
      onChange="foo"
      type="checkbox"
    />
  </div>
  <div
    className="todo-list-item__text"
  >
    <DraftEditorBlock
      blockProps={
        Object {
          "checked": true,
          "onToggle": "foo",
        }
      }
      offsetKey="foo"
    />
  </div>
</div>
`;
